src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java

```diff
@@ -0,0 +1,210 @@
+package nl.hsac.fitnesse.fixture.slim;
+
+import nl.hsac.fitnesse.fixture.slim.exceptions.CouldNotFindMessageException;
+import nl.hsac.fitnesse.fixture.slim.exceptions.RuntimeIOException;
+import nl.hsac.fitnesse.fixture.slim.exceptions.RuntimeMessageException;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.mail.*;
+import javax.mail.internet.MimeMultipart;
+import javax.mail.search.SearchTerm;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.lang.String.format;
+
+public class EmailFixture extends SlimFixture {
+    private final static Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+    private static final Date HOURS_BACK = new Date(System.currentTimeMillis() - 1000 * 60 * 60 * 24);
+    private Store store;
+
+    public EmailFixture() {
+
+    }
+
+    /**
+     * | set mail provider with host | <i>host</i> | port | <i>port</i> | user | <i>username</i> | password | <i>password</i> |
+     */
+    public void setMailProviderWithHostPortUserPassword(final String host, final String port, final String username, final String password) {
+        final Properties props = new Properties();
+        try {
+            final Session session = Session.getDefaultInstance(props, null);
+            store = session.getStore("imaps");
+            store.connect(host, username, password);
+        } catch (final MessagingException e) {
+            throw new StopTestException("Cannot connect to mailserver");
+        }
+    }
+
+    /**
+     * | ensure | mail received with subject | <i>subject</i> | from | <i>receiver</i> |
+     *
+     * @param subject
+     * @return
+     */
+    public String mailReceivedWithSubjectFrom(final String subject, final String receiver) {
```

Comment:
a mail received from, I would expect the parameter to be the sender, not the reciever....? 

But given the code I see below I expect this method should probably be called: `mailReceivedByWithSubject` and the order of the parameters reversed. (Given that the receiver is likely to be shorter it is also good practice to have that as first parameter, to make rows easier to read)

Reply:
good one! Like this we can easily use email aliases